projects/ngx-translate-router/src/lib/localize-router.service.ts

```diff
@@ -178,19 +185,31 @@ export class LocalizeRouterService {
       const previousLang = this.parser.getLocationLang(previousEvent.url) || this.parser.defaultLang;
       const currentLang = this.parser.getLocationLang(currentEvent.url) || this.parser.defaultLang;
 
-      if (currentLang !== previousLang) {
-        // mutate router config directly to avoid getting out of sync
-        this.parser.mutateRouterRootRoute(currentLang, previousLang, this.router.config);
+      if (currentLang !== previousLang && this.latestUrl !== currentEvent.url) {
+        this.latestUrl = currentEvent.url;
+        this.cancelCurrentNavigation();
         this.parser.translateRoutes(currentLang)
-          .pipe(
-            // reset routes again once they are all translated
-            tap(() => this.router.resetConfig(this.parser.routes))
-          )
           .subscribe(() => {
+            // Reset routes again once they are all translated
+            this.router.resetConfig(this.parser.routes);
+            // Init new navigation with same url to take new congif in consideration
+            this.router.navigateByUrl(currentEvent.url);
             // Fire route change event
             this.routerEvents.next(currentLang);
           });
       }
+      this.latestUrl = currentEvent.url;
     };
   }
+
+  /**
+   * Drop the current Navigation
+   */
+  private cancelCurrentNavigation() {
+    const currentNavigation = this.router.getCurrentNavigation();
+    const url = this.router.serializeUrl(currentNavigation.extractedUrl);
+    (this.router.events as Subject<Event>).next(new NavigationCancel(currentNavigation.id, url, ''));
+    (this.router as any).transitions.next({...(this.router as any).transitions.getValue(), id: 0});
```

Comment:
@giacomo What to you think about this ?
Can it have impact on common project ? Is there somebody that use the transaction id ?

Reply:
do you mean the transitions for animations?

Reply:
The transition is the state that is created using a `NavigationStart` event.
The `NavigationCancel` has no effect here but I totally dropped the current transaction to cancel current navigation.

Reply:
I dont kwon if anyone is using transaction ids? but why dropping the current transaction, and why canceling the navigation? transitions are marked  as *private readonly*. a route is changed only in the package if the language changes. i check out the pr later and try to understand. IMHO we should not navigate to some routes, Angular is doing this job for.

Reply:
In new version of Angular 8, the previous transaction will not be dropped by new navigation. So the previous navigation will be executed with wrong route config.
I canceled the previous navigation to execute a new one with good configuration

Reply:
@gilsdav after check out the PR looks fine. However i could not reproduce an error with old version by clicking browser back!

Reply:
Back bug reproduction:

- Go to http://localhost:4200/en/mytest/george
- Click on "change" button to change lang
- Refresh the page
- Click on back button

The bug is when translation is not already fetched when back button is clicked

Reply:
@gilsdav ok know i got it ... i did not seen the error while using ssr both languages are loaded before clicking back ;) 

could i merge this?